### PR TITLE
added FungibleTokens contract (plural!)

### PR DESCRIPTION
**Pull request by justjoolz**
Added FungibleTokens contract 

https://github.com/onflow/flow-ft/issues/59

### CODE REVIEW

File: contracts/ExampleTokens.cdc [hunk 1]
```diff
@@ -0,0 +1,255 @@
+import FungibleTokens from "./FungibleTokens.cdc" // 0xFungibleTokensADDRESS
+
+pub contract ExampleTokens: FungibleTokens {
+
+    /// Total supply of ExampleTokenSets in existence
+    pub var totalSupplyByID: {UInt64: UFix64}
```
Comment (joshuahannan) [inline contracts/ExampleTokens.cdc:6]:
```suggestion
    access(self) var totalSupplyByID: {UInt64: UFix64}
```
dictionary fields should be private

File: contracts/ExampleTokens.cdc [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        pub fun burnTokens(from: @FungibleTokens.TokenVault) {
+            let vault <- from as! @ExampleTokens.TokenVault
+            let amount = vault.balance
+            let tokenID = vault.tokenID
+            destroy vault
+            emit TokensBurned(tokenID: tokenID, amount: amount)
+        }
+    }
+
+    pub resource Collection: FungibleTokens.CollectionPublic, FungibleTokens.CollectionPrivate {
```
Comment (joshuahannan) [inline contracts/ExampleTokens.cdc:196]: I love the idea of having collections for multiple fungible tokens. I think we'll definitely want to integrate something like this in the standard

**Review by joshuahannan — COMMENTED** [review]
Thanks for proposing this! Great suggestions. Like I said in the issue, we have a rough draft of a more fundamental change to the standard that we are going to propose soon, so this might not be needed, but if the community decides that we don't want to have to go through managing all those breaking changes, then this might be a great start for a alternate solution.

File: contracts/ExampleTokens.cdc [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+    }
+
+    pub fun createEmptyCollection(): @FungibleTokens.Collection {
+        return <- create Collection()
+    }
+
+    init() {
+        self.totalSupplyByID = {0: 1000.0}
+        self.CollectionStoragePath = /storage/ExampleTokens
```
[the remark this answers is not in the document]
Comment (justjoolz) [reply contracts/ExampleTokens.cdc:232]: Are you referring to `CollectionStoragePath` or the example storage path?

File: contracts/FungibleTokens.cdc [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+                self.tokenID == tokenID:
+                    "TokenID must be initalized to the supplied tokenID"
+            }
+        }
+    }
+
+
+    // Requirement that all conforming TokenVault smart contracts have
+    // to define a resource called TokenVault that conforms to Provider, Receiver, Balance
+    pub resource TokenVault: Provider, Receiver, Balance {
```
[the remark this answers is not in the document]
Comment (justjoolz) [reply contracts/FungibleTokens.cdc:139]: How would that work? (they return different Types and we can't use the existing Vault as it has no ID)

File: contracts/ExampleTokens.cdc (hunk 2 above)
Comment (justjoolz) [reply contracts/ExampleTokens.cdc:196]: I basically tried to create something that follows existing patterns from FT/NFT interfaces as there has been very little talk of updating the existing standard and the longer they are left in their current form the greater the friction breaking changes will introduce.

File: contracts/FungibleTokens.cdc (hunk 4 above)
[the remark this answers is not in the document]
Comment (justjoolz) [reply contracts/FungibleTokens.cdc:139]: My idea is to introduce this as a new standard not to replace the existing one, so anyone who wants to support these tokens could integrate the complementary 'plural' standard.

Main drawback with breaking change to FT I see is how long will it take for exchanges to update? They already struggle with the smallest change....

**Pull request closed without merging**